### ISSUE TITLE
Manage users modals do not close on `esc` key

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#4028]
## Area or Component
This issue is seen application wide, but this ticket is specifically for the modals in manage users (reset password, add new user, reset MFA, etc.)

## Quick Summary
The modals do not respond to the `esc` key and instead need to be exited with a button press.

### How to Replicate
1. Open a modal (like reset password)
2. Tab through the modal a bit
3. Try to exit using only the keyboard
4. See that esc is not an option and you have to tab forward/backward to the "close" instead

## Accessibility Violation
ESC key does not close modal: https://cdc-devtools.dequecloud.com/rule-help/en/modal-no-esc

## Action Requested
Change the modal component to respond to the `esc` key.

## Source Code (optional)
There are a few different files that will need to be changed. It looks like we're largely using React's built-in `Modal` component, which will apparently close on ESC give an `onRequestClose` prop: https://github.com/reactjs/react-modal/issues/314

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=1647999156&range=18:18
